controllers/jujucluster_controller.go

```diff
@@ -198,72 +201,68 @@ func (r *JujuClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 					return ctrl.Result{}, err
 				}
 				if model != nil {
-					if model.Status.Status != "destroying" {
-						modelUUID, err := jujuClient.Models.GetModelUUID(ctx, jujuCluster.Name)
-						if err != nil {
-							log.Error(err, "failed to retrieve modelUUID when attempting to destroy model")
-							return ctrl.Result{}, err
-						}
-
-						if modelUUID != "" {
-							log.Info("destroying model")
-							destroyModelInput := juju.DestroyModelInput{
-								UUID: modelUUID,
-							}
-							if err := jujuClient.Models.DestroyModel(ctx, destroyModelInput); err != nil {
-								log.Error(err, "failed to destroy model")
+					if model.Life == life.Alive {
+						if model.Status.Status != "destroying" {
+							modelUUID, err := jujuClient.Models.GetModelUUID(ctx, jujuCluster.Name)
+							if err != nil {
+								log.Error(err, "failed to retrieve modelUUID when attempting to destroy model")
 								return ctrl.Result{}, err
 							}
-							log.Info("request to destroy model succeeded, requeueing")
-							return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+
+							if modelUUID != "" {
```

Comment:
yeah it could be helpful. Before I increased the requeue time (I think previously I was requeuing immediately) I was seeing issues where the model would report as alive and active, but the model uuid was coming back as empty since I guess theres a delay in updating model status and the reconcile loop was going too fast. Would be nice to at least log when this happens, maybe not an error but a warning might be better (something like I wanted to delete this but it looks like its already gone)). Will add